subdomains/blog/_posts/2020-11-25-command-spotlight-volta-run.md

```diff
@@ -0,0 +1,42 @@
+---
+layout: post
+title:  "Command Spotlight: volta run"
+date:   2020-11-23 08:00:00 -0700
+categories: [spotlight, commands]
+author: Chuck Pierce
+excerpt_separator: <!--more-->
+---
+
+Volta's context detection is designed to be invisible: You type a command—like `npm start`—and Volta automatically detects and runs the correct version of that command. Sometimes, however, you need to take direct control over the tool versions that are used. Maybe you are trying to reproduce reported bug or testing how your project works with different tool versions. Whatever the reason, **`volta run` lets you run any command while specifying the Node, npm, and Yarn versions right at the command line**
+<!--more-->
+
+[`volta run`](https://docs.volta.sh/reference/run) has a few sets of command-line flags that let you define your tool versions. For any tool that _doesn't_ have a version set in the command line flags, Volta will use its normal context resolution to determine the appropriate version (from `package.json` or your default versions, as needed). The flags that you can set are:
+
+- `--node <version>` controls the Node version for the command. `<version>` here can be an exact version—like `14.15.1`—or a partial version—such as `12`. You can even use `latest` or `lts` to resolve the appropriate Node version.
+- `--npm <version>` and `--bundled-npm` allow you to set the npm version. Just like for Node, `<version>` can be an exact or partial version. `--bundled-npm` tells Volta to ignore any specific `npm` settings in the project or defaults and instead use the version bundled with Node (whichever Node you are using).
+- `--yarn <version>` and `--no-yarn` similarly allow you to set the Yarn version. Since it isn't bundled, `--no-yarn` tells Volta to ignore any settings and omit Yarn from the final environment entirely.
+- `--env NAME=VALUE` lets you set an environment variable for the command as well. This can be set multiple times to set multiple different environment variables.
+
+That's a lot of words, how about some examples?
```

Comment:
I feel like you should start with some examples. That's the most compelling part. I'm not even sure you need to get into the details above that. That's more like documentation.

Reply:
Yeah I think the post actually just gets more approachable if you drop the details section above. Towards the bottom of the post you can direct them to the docs for more information.

Reply:
Agreed, that works a lot better, especially since the docs already lay out everything that was rehashed there :+1: